### PR TITLE
Modified the default label text box into a drop down

### PR DESCRIPTION
**Pull request by adlokib**
I have changed the text box for the default label check box into a drop down list which takes the classes from predefined_classes.txt so that new (misspelled) class names do not get added by mistake. If new classes are to be added, they can be added in predefined_classes.txt. Hence reducing the likelihood of error in using the feature.

labelimg_1

labelimg_2

### CODE REVIEW

File: libs/default_label_combobox.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    # Ref:
+    # http://pyqt.sourceforge.net/Docs/PyQt4/incompatible_apis.html
+    # http://stackoverflow.com/questions/21217399/pyqt4-qtcore-qvariant-object-instead-of-a-string
+    if sys.version_info.major >= 3:
+        import sip
+        sip.setapi('QVariant', 2)
+    from PyQt4.QtGui import QWidget, QHBoxLayout, QComboBox
+
+
+class default_label_ComboBox(QWidget):
```
Comment (tzutalin) [inline libs/default_label_combobox.py:15]: could change the naming style for class? it should start with capital

File: labelImg.py [hunk 2]
```diff
@@ -937,7 +944,7 @@ def new_shape(self):
                 text = self.label_dialog.pop_up(text=self.prev_label_text)
                 self.lastLabel = text
         else:
-            text = self.default_label_text_line.text()
+            text = self.default_label
```
Comment (tzutalin) [inline labelImg.py:947]: I forgot the difference between  self.default_label_text_line.text()   and self.default_label? Could you elaborate it?

Comment (adlokib) [reply labelImg.py:947]: Earlier, if default label checkbox was ticked, the class name was extracted from the text-box, this is self.default_label_text_line.text(). 

Now, that it has been converted to a drop-down list, I am setting the value of self.default_label, whenever it is changed. On application startup, it is initialized with the first class from predefined_classes.txt

self.default_label was not present earlier, its a class attribute I have introduced. its only purpose is to hold the selection of the default label drop down.

File: libs/default_label_combobox.py (hunk 1 above)
Comment (adlokib) [reply libs/default_label_combobox.py]: I have made the change.

**Review by tzutalin — COMMENTED** [review]
Thanks

**Pull request merged**